modules/navier_stokes/src/postprocessors/INSElementIntegralEnergyAdvection.C

```diff
@@ -0,0 +1,52 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSElementIntegralEnergyAdvection.h"
+#include "NS.h"
+
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("NavierStokesApp", INSElementIntegralEnergyAdvection);
+registerMooseObject("NavierStokesApp", INSADElementIntegralEnergyAdvection);
+
+template <bool is_ad>
+InputParameters
+INSElementIntegralEnergyAdvectionTempl<is_ad>::validParams()
+{
+  InputParameters params = ElementIntegralPostprocessor::validParams();
+  params.addClassDescription("Computes the volume integrated energy transport due to advection");
```

Comment:
this is not the "transport" that s not a quantity (even though I understand what you mean)

This is more like the contribution of advection to the energy balance. 

Computes the net volumetric balance of energy transported by advection ??

Reply:
I think that sounds reasonable